### PR TITLE
Retry after HTTP timeout

### PR DESCRIPTION
**Pull request by AnastasiaKubova**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests if this modifies the Windows code?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
## Related PRs or issues

#1361
[AB#78672](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/78672)

### CODE REVIEW

File: SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Channel/Channel.cs [hunk 1]
```diff
@@ -453,7 +453,7 @@ private void HandleSendingResult(State state, string batchId, IServiceCall call)
                 }
                 else if (call.IsFaulted)
                 {
-                    AppCenterLog.Error(AppCenterLog.LogTag, $"Sending logs for channel '{Name}', batch '{batchId}' failed: {call.Exception?.Message}");
+                    AppCenterLog.Error(AppCenterLog.LogTag, $"Sending logs for channel '{Name}', batch '{batchId}' failed: {call.Exception?.Message} \n{call.Exception?.InnerException}");
```
Comment (MatkovIvan) [inline SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Channel/Channel.cs:456]:
```suggestion
                    AppCenterLog.Error(AppCenterLog.LogTag, $"Sending logs for channel '{Name}', batch '{batchId}' failed: {call.Exception?.ToString()}");
```

File: SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Ingestion/Http/HttpNetworkAdapter.cs [hunk 2]
```diff
@@ -17,6 +16,7 @@ public sealed partial class HttpNetworkAdapter : IHttpNetworkAdapter
         internal const string ContentTypeValue = "application/json; charset=utf-8";
 
         private HttpClient _httpClient;
+        private static HttpClient _customHttpClient;
```
Comment (MatkovIvan) [inline SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Ingestion/Http/HttpNetworkAdapter.cs:19]: Remove it. You can pass client to constructor for tests

File: SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Ingestion/Http/HttpNetworkAdapter.cs [hunk 3]
```diff
@@ -125,6 +137,10 @@ private async Task<HttpResponseMessage> SendRequestAsync(HttpRequestMessage requ
             {
                 return await HttpClient.SendAsync(request, cancellationToken).ConfigureAwait(false);
             }
+            catch (HttpRequestException e)
```
Comment (MatkovIvan) [inline SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Ingestion/Http/HttpNetworkAdapter.cs:140]: Are you sure that we should catch all these exception without additional conditions?

File: Tests/Microsoft.AppCenter.Test.Windows/Channel/ChannelTest.cs [hunk 4]
```diff
@@ -560,5 +591,9 @@ private static int EventWithSemaphoreOccurred(SemaphoreSlim semaphore)
                 }
             }
         }
+
+        public class MockHttpClient : HttpClient
```
Comment (MatkovIvan) [inline Tests/Microsoft.AppCenter.Test.Windows/Channel/ChannelTest.cs:595]: Why do you need this class? Can you mock `HttpClient` directly?

File: SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Ingestion/Http/HttpNetworkAdapter.cs [hunk 5]
```diff
@@ -125,6 +132,10 @@ private async Task<HttpResponseMessage> SendRequestAsync(HttpRequestMessage requ
...
+            {
+                throw new NetworkIngestionException(e);
```
Comment (guperrot) [inline SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Ingestion/Http/HttpNetworkAdapter.cs:137]: So how does 4XX vs 5XX work then?

Comment (MatkovIvan) [reply SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Ingestion/Http/HttpNetworkAdapter.cs:137]: All server answers with any codes does not throw this exception (there is an API to do it, but not by default). Handling error response codes is not changed

Comment (guperrot) [reply SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Ingestion/Http/HttpNetworkAdapter.cs:137]: Do we know the exact conditions when this is thrown then?

Comment (MatkovIvan) [reply SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Ingestion/Http/HttpNetworkAdapter.cs:137]: Yes, for example SSL errors.
The code for testing:
```C#
AppCenter.SetLogUrl("https://self-signed.badssl.com");
```

Comment (MatkovIvan) [reply SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Ingestion/Http/HttpNetworkAdapter.cs:137]: From original issue it was `The SSL connection could not be established, see inner exception.`, probably timeout on handshake or so. Anyway, this type of exceptions is intended to be retried

**Pull request merged**